### PR TITLE
Revamp Role Permissions as Scope Assignments

### PR DESCRIPTION
**Pull request by RakhithaRR**
This PR revamps the Role Permissions UI as Scope Assignments to avoid confusions with the role permission settings in the management console.

- Changed feature title
- Changed button and label texts
- Updated other terminology to scope assignment/mapping related terms
- Added a warning message when adding a new scope mapping with either alias or scope assignments

Fixes https://github.com/wso2/product-apim/issues/9572

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx [hunk 1]
```diff
@@ -40,10 +40,13 @@ const headCells = [
         numeric: false, disablePadding: false, label: 'Roles',
     },
     {
-        id: 'permissions', numeric: false, disablePadding: false, label: 'Permissions',
+        id: 'permissions', numeric: false, disablePadding: false, label: 'Scope Assignments',
     },
 ];
 
+const pageDesc = 'Scope assignments are related to internal APIM specific scope assignments and are not related'
+                + ' to role permission assignments in the management console.';
```
Comment (ruthryi) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx:48]: Scope assignments are only related to internal, APIM-specific scope assignments. They are not related to role permission assignments in the Management Console.

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/site/public/locales/en.json [hunk 2]
```diff
@@ -332,9 +332,9 @@
   "Manage.Alerts.unsubscribe.confirm.dialog.message": "This will remove all the existing alert subscriptions and emails. This action cannot be undone.",
   "Manage.Alerts.unsubscribe.error.msg": "Error occurred while unsubscribing.",
   "Manage.Alerts.unsubscribe.success.msg": "Unsubscribed from all alerts successfully.",
-  "RolePermissions.ListRoles.permission.denied.content": "You dont have enough permission to view Role Permissions. Please contact the site administrator.",
+  "RolePermissions.ListRoles.permission.denied.content": "You dont have enough permission to view Scope Assignments. Please contact the site administrator.",
```
Comment (ruthryi) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/site/public/locales/en.json:335]: "don't" or "do not"

Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/site/public/locales/en.json]: Fixed with 79a0036

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx (hunk 1 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx]: Fixed with 79a0036

**Review by ruthryi — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx [hunk 3]
```diff
@@ -232,15 +235,15 @@ export default function ListRoles() {
...
+        <ContentBase title='Scope Assignments' pageDescription={pageDesc}>
             <ListAddOns>
                 <Grid item>
                     <Button
                         variant='contained'
                         color='primary'
                         onClick={() => setIsOpen(true)}
                     >
-                        Add role permission
+                        Add scope mappings
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx:246]: Missing i18n definition.

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx [hunk 4]
```diff
@@ -118,7 +118,7 @@ export default function PermissionsSelector(props) {
                     <Typography variant='h5' display='block' gutterBottom>
                         {role}
                         <Box display='inline' pl={1}>
-                            <Typography variant='caption' gutterBottom>Select Permissions</Typography>
+                            <Typography variant='caption' gutterBottom>Select Scope Assignments</Typography>
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx:121]: Missing i18n

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx [hunk 5]
```diff
@@ -104,7 +104,7 @@ export default function PermissionsSelector(props) {
                 variant='outlined'
                 color='primary'
             >
-                Permissions
+                Scope Assignments
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx:107]: Missing i18n

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx [hunk 6]
```diff
@@ -124,15 +124,15 @@ export default function AddRoleWizard(props) {
...
+                            Added scope mapping for
                             <b>{` ${newRole} `}</b>
                             successfully
                         </span>,
                     );
                     onClose();
                 })
                 .catch((error) => {
-                    Alert.error('Something went wrong while adding new role permissions');
+                    Alert.error('Something went wrong while adding new scope mapping');
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx:135]: Missing i18n

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx [hunk 7]
```diff
@@ -171,8 +171,8 @@ export default function AddRoleWizard(props) {
         <AddItem
             onSave={onAddRole}
             onClose={onClose}
-            title='Add new role permissions'
-            buttonText='Add role permissions'
+            title='Add new scope mapping'
+            buttonText='Add scope mapping'
```
Comment (menakaj) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx:175]: Missing i18n

Comment (menakaj) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx:175]: Please find and fix in the other places accordingly as well

**Review by menakaj — CHANGES_REQUESTED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx (hunk 7 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx]: Fixed with 46a151e

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx (hunk 6 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/Commons/AddRoleWizard.jsx]: Fixed with 46a151e

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx (hunk 5 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx]: Fixed with 46a151e

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx (hunk 4 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/TreeView/PermissionsSelector.jsx]: Fixed with 46a151e

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx (hunk 3 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/ListRoles.jsx]: Fixed with 46a151e

**Review by menakaj — APPROVED** [review]

**Comment by bhathiya** [comment]
@RakhithaRR why is this still open?

**Pull request merged**